firmware/boron-ins-fsm/src/consoleFunctions.cpp

```diff
@@ -268,4 +269,32 @@ int im21_door_id_set(String command) { // command is a long string with all the
 
   return 1;
 
+}
+
+int force_reset(String command){
+
+  //default to invalid input
+  int returnFlag = -1;
+
+  //string.toInt() returns 0 if it fails, so can't distinguish between user 
+  //entering 0 vs entering bad input. So convert to char and use ascii table
+  const char* holder = command.c_str();
+
+  if(*(holder+1) != 0){
+    //any string longer than 1 char is invalid input, so
+    returnFlag = -1;
+  }
+  else if(*holder == '1'){
+    returnFlag = 1;
+    bool msg_sent = Particle.publish("YOU SHALL NOT PANIC!!", "Reset has begun so ignore the future particle message about failure to call force_reset()", PRIVATE | WITH_ACK);
+    if (msg_sent)
+      System.reset();
```

Comment:
I am personally of the opinion that in C-derived languages, one should put curly braces around the bodies of conditionals and loops. I have two main reasons:
1. If you want to add something to the body of the loop/conditional, then you might forget to add the braces and then your code will be wrong even if the indentation makes it look correct. So difficult to debug
2. If you want to add something to the body of the loop/conditional and you do remember to add the curly braces, then in my preferred style that would add the opening brace on the same line as the if/while/for/etc which would make it show up in the git diff when really nothing there changed.

That being said, I don't change the firmware enough to know what the typical style here is. So I would defer to what is normal in the rest of the code.